src/server/scripts/Commands/cs_pooltools.cpp

```diff
@@ -0,0 +1,409 @@
+                }
+            }
+
+            if (existingGroup)
+            {
+                existingGroup->FoundObjects.push_back({ go->GetEntry(), spawnId });
+                addedCount++;
+            }
+            else
+            {
+                NodeGroup newGroup;
+                newGroup.X = go->GetPositionX();
+                newGroup.Y = go->GetPositionY();
+                newGroup.Z = go->GetPositionZ();
+                newGroup.FoundObjects.push_back({ go->GetEntry(), spawnId });
+
+                session.CapturedGroups.push_back(newGroup);
+                newGroupsCount++;
+                addedCount++;
+            }
+        }
+
+        if (addedCount == 0)
+        {
+            handler->SendErrorMessage("No new matching objects found in {}y radius.", radius);
+            return false;
+        }
+
+        handler->PSendSysMessage("|cff00ff00Scan Complete.|r Added {} objects into {} new groups.", addedCount, newGroupsCount);
+
+        if (!session.CapturedGroups.empty())
+        {
+            NodeGroup& lastGroup = session.CapturedGroups.back();
+            for (auto& p : lastGroup.FoundObjects)
+            {
+                handler->PSendSysMessage(" - Entry {} (GUID: {})", p.first, p.second);
+            }
+        }
+        return true;
+    }
+
+    static bool HandlePoolRemove(ChatHandler* handler)
+    {
+        ObjectGuid playerGuid = handler->GetPlayer()->GetGUID();
+        if (sPoolSessions.find(playerGuid) == sPoolSessions.end())
+        {
+            handler->SendErrorMessage("No active session.");
+            return false;
+        }
+
+        PoolSession& session = sPoolSessions[playerGuid];
+
+        if (session.CapturedGroups.empty())
+        {
+            handler->SendErrorMessage("No groups captured.");
+            return false;
+        }
+
+        NodeGroup removed = session.CapturedGroups.back();
+        session.CapturedGroups.pop_back();
+
+        handler->PSendSysMessage("|cffff0000Undo Successful.|r Removed last group containing {} objects.", removed.FoundObjects.size());
+        return true;
+    }
+
+    static bool HandlePoolEnd(ChatHandler* handler)
+    {
+        ObjectGuid playerGuid = handler->GetPlayer()->GetGUID();
+        auto it = sPoolSessions.find(playerGuid);
+        if (it == sPoolSessions.end()) return false;
+
+        PoolSession& session = it->second; // Use reference from iterator
+
+        auto EscapeSQL = [](std::string_view input) -> std::string {
+            std::string safe;
+            safe.reserve(input.size());
+            for (char c : input) {
+                if (c == '\'') safe += "\\'";
+                else safe += c;
+            }
+            return safe;
+            };
+
+        bool complexPool = (session.CurrentTemplate.size() > 1);
+
+        // SQL Variables and Header
+        LOG_DEBUG("sql.dev", "-- Pool Dump: {}\n", session.ZoneName);
+        LOG_DEBUG("sql.dev", "SET @mother_pool := @mother_pool+1;\n");
```

Comment:
I think the line break here is not needed

Reply:
oh yeah just noticed in the test I made lol